src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+using Microsoft.VisualStudio.Services.CircuitBreaker;
+using Microsoft.WorkItems;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public class SarifWorkItemFiler
+    {
+        public SarifWorkItemFiler() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SarifWorkItemFiler"> class.</see>
+        /// </summary>
+        /// <param name="filingClient">
+        /// A client for communicating with a work item filing host (for example, GitHub or Azure DevOps).
+        /// </param>
+        /// <param name="filingContext">
+        /// A root context object that configures the work item filing operation.
+        /// </param>
+        public SarifWorkItemFiler(SarifWorkItemContext filingContext)
+        {
+            this.FilingContext = filingContext ?? throw new ArgumentNullException(nameof(filingContext));
+            this.FilingClient = FilingClientFactory.CreateFilingTarget(filingContext.ProjectUri);
+        }
+
+        public SarifWorkItemContext FilingContext { get; set; }
+
+        public FilingClient FilingClient { get; set; }
+
+        public virtual void FileWorkItems(Uri sarifLogFileLocation)
+        {
+            sarifLogFileLocation = sarifLogFileLocation ?? throw new ArgumentNullException(nameof(sarifLogFileLocation));
+
+            if (sarifLogFileLocation.IsAbsoluteUri && sarifLogFileLocation.Scheme == "file:")
```

Comment:
>file: [](start = 86, length = 5)

Should be just "file" without the colon to match file URI's correctly.

Reply:
"Mile wide hole in unit tests" is the phrase you are probably reaching for. :) Somewhere in .NET (in one of the web assemblies) someone declares a string constant for this and I always have trouble locating it.

---
In reply to: [381535789](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381535789) [](ancestors = 381535789)